### ISSUE TITLE
Unexpected result for compose Multiply

### ISSUE THREAD

**Issue by Danack** [ImageMagick/ImageMagick#4448]
### ImageMagick version

7.1.0-13

### Operating system

Linux

### Operating system, version and so on

Debian 11

### Description

Using compose and the Multiply option seems to give a bad result on ImageMagick 7, or at least very different from the result on ImageMagick 6.  This affects both command line utilities and through the MagickWand C api. Example outputs are below.

I can't see anything in the upgrade notes about a change to the compose option.

### Steps to Reproduce

ImageMagick 6 command

```
composite -compose Multiply top.png bottom.png output_magick_im6.png
```

ImageMagick 7 command
```
magick composite -compose Multiply top.png bottom.png output_magick_im7.png
```

In case it's of use, this is the result of identify for the source images.
```
identify top.png
# top.png PNG 1041x785 1041x785+0+0 8-bit Grayscale Gray 4104B 0.000u 0:00.002
identify bottom.png
# bottom.png PNG 1053x785 1053x785+0+0 8-bit Grayscale Gray 382330B 0.000u 0:00.006
```

### Images

top.png
![top](https://user-images.githubusercontent.com/1505719/140610550-53eb3f79-35b1-4207-88cd-b8892d2d088d.png)

bottom.png
![bottom](https://user-images.githubusercontent.com/1505719/140610564-e3b3e043-0a54-4c4f-b5a6-83fe602a641e.png)

ImageMagick6 result
![output_magick_im6](https://user-images.githubusercontent.com/1505719/140610587-07cd6f71-52b9-48b5-87e0-b1728611d433.png)

ImageMagick7 result
![output_magick_im7](https://user-images.githubusercontent.com/1505719/140610603-2bbedecc-69cb-486d-9657-6a06afc96d61.png)

**Comment by snibgo** [comment]
V7 handles alpha differently. See http://www.imagemagick.org/script/porting.php . To replicate v6 behaviour, insert "-channel RGB,sync" before doing the composite.

I do not recommend "magick composite...", because it is less flexible than "magick inA inB -composite out"

However, this doesn't totally solve the problem. It seems IM has a bug. "-compose Multiply -composite" should give the same result whatever order the inputs are given, but it doesn't. For example, with IM v7.1.0-4 on Windows 8.1:

```
f:\web\im>%IMG7%magick xc:graya(10%,1) xc:graya(20%,0) -channel RGB,sync -compose Multiply -composite txt:
# ImageMagick pixel enumeration: 1,1,65535,graya
0,0: (6553.5,65535)  #199A199A199AFFFF  graya(10%,1)

f:\web\im>%IMG7%magick xc:graya(20%,0) xc:graya(10%,1) -channel RGB,sync -compose Multiply -composite txt:
# ImageMagick pixel enumeration: 1,1,65535,graya
0,0: (6553.5,0)  #199A199A199A0000  graya(10%,0)
```
A developer may look into this.

**Comment by urban-warrior** [comment]
ImageMagick 7 was designed in response to certain undesirable design decisions in IM V6.  One thing we wanted to get rid of was `sync` as it is not a real channel but instead a flag.  We will restore 'pure' mathematical operations in compositions with the preferred method, a define, `-define compose:sync=false`.

**Comment by Danack** [comment]
I'm not sure if the commit above means that I should be testing or not, but would it possible for you to tell me the exact command I should be running to get the equivalent output as ImageMagick 6?

I tried running what I think might be the correct commands after compiling from git head, but I don't seem to get output similar to the ImageMagick 6 output - see below.

Also, I orignally found this issue from a report of a [PHP Imagick user](https://github.com/Imagick/imagick/issues/485) before finding the command line seemed to have similar behaviour. Would the fix in 5d3bc2abc650c16eff55c7d89f3d49b819f733c1 be likely to fix the issue when going through the MagickWand api?

`magick composite -define compose:sync=true -compose Multiply top.png bottom.png output_magick_im7_sync_true.png`
![output_magick_im7_sync_true](https://user-images.githubusercontent.com/1505719/141680180-0714d381-e07a-483a-aaaa-7a4c852c1666.png)

`magick composite -define compose:sync=false -compose Multiply top.png bottom.png output_magick_im7_sync_false.png`
![output_magick_im7_sync_false](https://user-images.githubusercontent.com/1505719/141680191-50a43e09-089f-40bb-81cb-91a577b3e619.png)

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/download/beta/ by sometime tomorrow.
